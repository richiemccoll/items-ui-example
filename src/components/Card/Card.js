import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Button, Header } from "semantic-ui-react";

const Container = styled.div`
    width: 300px;
    margin: 0 auto;
    box-shadow: 0 12px 15px 0 rgba(0, 0, 0, 0.24), 0 17px 50px 0 rgba(0, 0, 0, 0.19);
    font-family: 'Open Sans', sans-serif;
    margin: .5em;
    padding: .5em;
`;

const ImageContainer = styled.div`
    width: 100%;
    height: 255px;
    position: relative;
`;

const Image = styled.img`
    width: 100%;
    height: 100%;
`;

const Content = styled.div`
    padding: 0 .5em 0 .5em;
`;

const Card = ({ id, title, image, description, toggleForm }) => {
  return (
    <Container>
      <ImageContainer>
        <Image src={image} alt={title} />
      </ImageContainer>
      <Content>
        <Header as="h2">
          {title}
        </Header>
        <Header as="h3">
          {description}
        </Header>
        {toggleForm &&
          <Button
            basic
            color="green"
            onClick={() =>
              toggleForm("edit", { id, title, image, description })}
          >
            Edit
          </Button>}
      </Content>
    </Container>
  );
};

Card.PropTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  toggleForm: PropTypes.func
};

export default Card;
