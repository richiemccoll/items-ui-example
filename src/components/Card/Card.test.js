import React from "react";
import Card from "./Card";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from 'sinon';
import chai from 'chai';

describe("Card Component", () => {
  const testCard = {
    id: 1,
    title: "Test Card",
    image: "",
    description: "test desc",
    toggleForm() {
      return;
    }
  };

  it("should render correctly based on props", () => {
    const component = renderer.create(
      <Card
        image={testCard.image}
        title={testCard.title}
        description={testCard.description}
        toggleForm={testCard.toggleForm}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call the toggleForm prop", () => {
    const toggleForm = sinon.spy();
    const wrapper = mount(
      <Card toggleForm={toggleForm} />
    );
    wrapper.find('button').simulate('click');
    chai.expect(toggleForm).to.have.property('callCount', 1);
  });
});
