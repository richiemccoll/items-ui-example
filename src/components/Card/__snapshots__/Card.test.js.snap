// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render correctly based on props 1`] = `
<div
  className="sc-bdVaJa iyRwOl"
>
  <div
    className="sc-bwzfXH gOHCgz"
  >
    <img
      alt="Test Card"
      className="sc-htpNat emEaka"
      src=""
    />
  </div>
  <div
    className="sc-bxivhb gutIee"
  >
    <h2
      className="ui header"
    >
      Test Card
    </h2>
    <h3
      className="ui header"
    >
      test desc
    </h3>
    <button
      className="ui green basic button"
      onClick={[Function]}
      tabIndex={undefined}
    >
      Edit
    </button>
  </div>
</div>
`;
